src/scenarios.py

```diff
@@ -0,0 +1,148 @@
+
+    Keyword Arguments: 
+           bat_range -- np.array() of battery capacities (kWh)
+    none_depot_range -- np.array() of non-depot charger power (kW)
+                rate -- interest rate for PMT calculation
+               years -- number of periods for PMT (years) 
+      bus_base_price -- floor price of a bus without any battery capacity (currency)
+          bprice_kwh -- Battery price per kWh, scales linearly
+          cprice_bos -- Estimated "balance of system" EVSE price/kW, scales 
+                        linearly. Default:$265/kW, see notes
+       cprice_scaled -- Estimated EVSE price/kW, derived from average costs,
+                        scales linearly. Default:$511/kW, see notes
+           bos_coeff -- weight to give to BOS EVSE price. Default: 0.8, see notes.
+        scaled_coeff -- weight to give to scaled avg EVSE price. Default: 0.2, 
+                        see notes.
+
+  --- NOTES: ---
+  This is derived from the work of M. McCall et al in https://doi.org/10.1088/1748-9326/ab560d
+  """
+  bat_pv            = bat_range * bprice_kwh
+  charger_pv_bos    = non_depot_range * cprice_bos
+  charger_pv_scaled = non_depot_range * cprice_scaled
+
+  annualized_base_bus        = -np.pmt(rate,years,bus_base_price)
+  annualized_bat             = -np.pmt(rate,years / 2,bat_pv) * 2# assumes a mid-life battery replacement.
+  annualized_charge_bos      = -np.pmt(rate,years,charger_pv_bos)
+  annualized_charge_scaled   = -np.pmt(rate,years,charger_pv_scaled)
+
+  anl_charge_weighted = (annualized_base_bus * bos_coeff) + (annualized_charge_scaled * scaled_coeff) 
+  results = {'annualized_base_bus':annualized_base_bus,
+             'annualized_bat':dict(zip(bat_range,annualized_bat)),
+             'annualized_charger':dict(zip(non_depot_range,anl_charge_weighted)),
+            }
+  return results
+
+def buildArgs():
+  parser = argparse.ArgumentParser(description='TODO')
+  parser.add_argument('parsed_gtfs_prefix',   type=str, help='TODO')
+  parser.add_argument('osm_data',             type=str, help='TODO')
+  parser.add_argument('depots_filename',      type=str, help='TODO')
+  parser.add_argument('output_dir',           type=str, help='TODO')
+  parser.add_argument('battery_cap_kwh',      type=str, help='TODO')
+  parser.add_argument('nondepot_charger_rate',type=str, help='TODO')
+  parser.add_argument('--sim-parameters',     type=str, help='TODO')
+  #TODO add charger_density when supported by sim.py
+  # parser.add_argument('-s','--save_csv',nargs='?',const='scenarios_results',type=str,help='TODO')
+  args = parser.parse_args()
+  return args
+
+def main(args=None):
+    #use for testing outside of CLI
+  if args is None:
+    args = buildArgs()
+  #extract scenario-related arguments and generate scenario array
+  list_args = set(['battery_cap_kwh','nondepot_charger_rate',])
+  #extract only the relevant arguments for building scenario matrix
+  scen_args = {k:v for (k,v) in vars(args).items() if k in list_args}
```

Comment:
This is a discomfortingly complicated way of doing things and won't raise errors on mismatches. I suggest being explicit instead.